packages/nextcloud/lib/src/api/webdav/webdav_client.dart

```diff
@@ -51,15 +52,26 @@ class WebDavClient extends DynamiteClient {
           authentications: client.authentications,
         );
 
-  Uri _constructUri([PathUri? path]) => constructUri(baseURL, path);
+  String? _username;
+
+  Future<Uri> _constructUri([PathUri? path]) async {
+    if (_username == null) {
+      final client = provisioning_api.$Client.fromClient(this);
+      // Consumers already need to catch http.ClientException so we don't need to catch and rethrow here.
+      final response = await client.users.getCurrentUser();
+      _username = response.body.ocs.data.id;
```

Comment:
So we could take the `loginName`  used by the root `NextcloudClient` as long as it was instantiated using the username and app password?
But because we also support instantiating it with a different loginName like email, we have to do this hack?

Reply:
No, here it has to be the username.
The problem is even with basic auth it is not guaranteed that it is the username, as you can also use loginName:password for that.
So sadly no optimization potential here.